### PR TITLE
Embedded ros2 tables

### PR DESCRIPTION
**Pull request by iluetkeb**
To better associate the various ingredients, make it tables with keys.

Also, introduce a couple of explicit questions and answers.

### CODE REVIEW

**Review by ralph-lange — APPROVED** [review]

File: articles/embedded.md [hunk 1]
```diff
@@ -21,26 +21,53 @@ While standard ROS 2 can run on Linux-class embedded devices, smaller microcontr
...
+To answer these questions, both the OFERA EU project as well as several others have already undertaken or are planning exploratory work.
+
+*Meta-Note*: Please only add a short description here, linking to more detailed pages if necessary.
 
+| Key | Related Question | Action | Description | Links |
+|-----|------------------|--------|-------------|-------|
+| A-RTOS | Q-RTOS | RTOS Proof-of-Concept | Provide a proof-of-concept RTOS. In the OFERA project, we chose NuttX, because it is largely POSIX compatible and thus eases porting. There are also experiments based on RIOT and FreeRTOS | TODO |
+| A-BUILD-META | Q-BUILD | Meta-Build | Explores a meta-build approach to transform ROS 2 CMakeLists.txt to RTOS-specific build instructions. | TODO |
+| A-BUILD-NUTTX | Q-BUILD | NuttX-specific build | OFERA has integrated (parts of) micro-ROS directly as an app in the NuttX build. |TODO |
+| A-BUILD-ANDROID | Q-BUILD | Arduino Build | Robotis has explored building all the libraries using the Arduino IDE. This required some manual changes and thus does not scale, but can get you off the ground. | TODO |
```
Comment (BorjaOuterelo) [inline articles/embedded.md]: `A-BUILD-ANDROID` should not it be `A-BUILD-ARDUINO`?

File: articles/embedded.md [hunk 2]
```diff
@@ -21,26 +21,53 @@ While standard ROS 2 can run on Linux-class embedded devices, smaller microcontr
...
+| W-MODULARITY | **High Modularity** | The embedded ROS 2 stack shall be highly modular, not only to allow for adaptation to specific hardware features and different RTOS, but also to allow the integration with existing frameworks and to be able to create customized sub-stacks. For example, it shall be usable in the style of rosserial, thus bringing basically the message serialization to the MCU only. Similarly, it should be possible to derive a sub-stack that provides node graph features and publish/subscribe but no parameters and services, and so on.
+| W-CONTROL | **Support control-oriented applications**  | MCUs are great for control, and micro-ROS should be as well. This usually means hard real-time performance with low jitter and small response times. |
+| W-MCU-SPEC | **MCU-specific core functionalities and mechanisms** | Finally, the embedded ROS 2 stack shall include novel core functionalities and mechanisms that are specific for MCUs. These include power management functionalities and mechanisms for static initialization and memory management.|
```
Comment (BorjaOuterelo) [inline articles/embedded.md]: If we plan to expand this table further, I suggest not use any linking word on it. For example, the `Finally,` will get out of place.

**Review by BorjaOuterelo — CHANGES_REQUESTED** [review]
I made a couple of comments.
I would add some contents to the tables on other PR as soon as this is merged.

File: articles/embedded.md (hunk 1 above)
Comment (iluetkeb) [reply articles/embedded.md]: Yep, done.

File: articles/embedded.md (hunk 2 above)
Comment (iluetkeb) [reply articles/embedded.md]: Good point. I just copied over the old content. Fixed now (also in other places).

**Review by BorjaOuterelo — APPROVED** [review]

**Pull request merged**